src/sympc/tensor/grads/grad_functions.py

```diff
@@ -520,39 +491,28 @@ def backward(ctx: Dict[str, Any], grad: MPCTensor) -> Tuple[MPCTensor]:
             grad (MPCTensor): The gradient that came from the child nodes
 
         Returns:
-            (input_grad, weight_grad) (Tuple[MPCTensor]): The gradients passed
+            (input_grad, weight_grad) (Tuple[MPCTensor, MPCTensor]): The gradients passed
             to the input and kernal nodes.
         """
-        input = ctx["input"]
+        x = ctx["x"]
         weight = ctx["weight"]
         stride = ctx["stride"]
         padding = ctx["padding"]
         dilation = ctx["dilation"]
         groups = ctx["groups"]
         weight_size = (weight.shape[2], weight.shape[3])
-        in_channels = input.shape[1]
+        in_channels = x.shape[1]
         out_channels = grad.shape[1]
-        min_batch = input.shape[0]
-
-        # Gradient w.r.t input of the Conv.
-        common_args = [
-            tuple(input.shape),
-            stride,
-            padding,
-            weight_size,
-            dilation,
-            grad.session,
-        ]
-        args = [[el] + common_args for el in grad.share_ptrs]
-
-        shares = parallel_execution(
-            GradConv2d.get_grad_input_padding, grad.session.parties
-        )(args)
-        grad_input_padding = MPCTensor(shares=shares, session=grad.session)
-
-        output_padding_tensor = grad_input_padding.reconstruct()
-        output_padding_tensor /= grad.session.nr_parties
-        output_padding = tuple(output_padding_tensor.to(torch.int).tolist())
+        min_batch = x.shape[0]
+
+        output_padding = torch.nn.grad._grad_input_padding(
+            grad_output=torch.empty(grad.shape),
```

Comment:
`grad_output` param is only used to get the ```shape/dim``` there is no need

Reply:
hmm didn't understand this.

Reply:
grad_output parameter is only used for the shape so there is no need to really compute it